### ISSUE TITLE
dbt templater 'File %s was not found in dbt project'

### ISSUE THREAD

**Issue by NiallRees** [sqlfluff/sqlfluff#597]
Just testing out the latest commit on my work dbt project to identify any issues on a real project. This one just sprung up:

```
Traceback (most recent call last):
  File "/Users/niallwoodward/dev/pull_requests/sqlfluff_venv/bin/sqlfluff", line 11, in <module>
    load_entry_point('sqlfluff', 'console_scripts', 'sqlfluff')()
  File "/Users/niallwoodward/dev/pull_requests/sqlfluff_venv/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/niallwoodward/dev/pull_requests/sqlfluff_venv/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/niallwoodward/dev/pull_requests/sqlfluff_venv/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/niallwoodward/dev/pull_requests/sqlfluff_venv/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/niallwoodward/dev/pull_requests/sqlfluff_venv/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/niallwoodward/dev/pull_requests/sqlfluff/src/sqlfluff/cli/commands.py", line 323, in lint
    ignore_files=not disregard_sqlfluffignores,
  File "/Users/niallwoodward/dev/pull_requests/sqlfluff/src/sqlfluff/core/linter.py", line 1161, in lint_paths
    ignore_files=ignore_files,
  File "/Users/niallwoodward/dev/pull_requests/sqlfluff/src/sqlfluff/core/linter.py", line 1139, in lint_path
    target_file.read(), fname=fname, fix=fix, config=config
  File "/Users/niallwoodward/dev/pull_requests/sqlfluff/src/sqlfluff/core/linter.py", line 939, in lint_string
    parsed = self.parse_string(in_str=in_str, fname=fname, config=config)
  File "/Users/niallwoodward/dev/pull_requests/sqlfluff/src/sqlfluff/core/linter.py", line 705, in parse_string
    in_str=in_str, fname=fname, config=config
  File "/Users/niallwoodward/dev/pull_requests/sqlfluff/src/sqlfluff/core/templaters/dbt.py", line 181, in process
    return self._unsafe_process(fname, in_str, config)
  File "/Users/niallwoodward/dev/pull_requests/sqlfluff/src/sqlfluff/core/templaters/dbt.py", line 220, in _unsafe_process
    raise RuntimeError("File %s was not found in dbt project" % fname)
RuntimeError: File /Users/niallwoodward/dev/tails-analytics-dbt/data_warehouse/deploy_post.sql was not found in dbt project
```

We have a query which is run by our dbt execution script called `deploy_post.sql`. Is the issue here that the dbt templater expects all .sql files to be part of the project?

**Comment by dmateusp** [comment]
Yes, that is what's happening.

What do you think would be a good fix? Does it work if you have a `.sqlfluff` file inside the directory that contains `deploy_post.sql` and pick the `jinja` templater there?

**Comment by NiallRees** [comment]
Just tried - doesn't work unfortunately. I think for now let's just document it as most people won't have this issue, and then revisit in the future if we decide to. I'll just sqlfluffignore the two scripts for now.

**Comment by NiallRees** [comment]
Let's also document that the directory `sqlfluff` is called from must be the root directory of the dbt project for dbt templating to work.

**Comment by NiallRees** [comment]
Resolved by #898
